### PR TITLE
7289 - remove radio NHP links

### PR DESCRIPTION
**Pull request by gmarinov**
updates network links logic to no longer link radio networks
adds tests for broadcast event presenter and for footer presenters in both design systems.

as per #1258 it also changes the footer logic to always display a network logo even if a network is not linked - see that PR for more details.

related: https://github.com/bbc/brandingtool-cloud/pull/180

### CODE REVIEW

File: src/Ds2013/Presenters/Domain/BroadcastEvent/broadcast_event.html.twig [hunk 1]
```diff
@@ -30,10 +30,11 @@
         </div>
         <div class="programme__service box-link__elevated micro text--subtle">
             {%- for breakdown in broadcast_event.getNetworkBreakdown -%}
+                {%- set net = breakdown.getNetwork() -%}
                 {% if not loop.first and not loop.last %},&#32;{%- endif -%}
-                {% if loop.last and loop.length > 1 %}&#32;&&#32;{%- endif -%}
-                {%- if breakdown.getNetwork() and network_link(breakdown.getNetwork()) -%}
-                    <a href="{{ network_link(breakdown.getNetwork()) }}">{{ breakdown.getNetworkName() }}</a>
+                {% if loop.last and loop.length > 1 %}&#32;&amp;&#32;{%- endif -%}
+                {%- if network_link(net) and not net.isRadio() -%}
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Domain/BroadcastEvent/broadcast_event.html.twig:36]: I think we need to include a check that net is defined/non-null here? The previous statement seems to be guarding against a null value. As do certain methods in the presenter class.

File: src/Ds2013/Presenters/Section/Footer/FooterPresenter.php [hunk 2]
```diff
@@ -32,11 +32,8 @@ public function getNetwork(): ?Network
 
     public function getNetworkUrlKey(): string
     {
-        // urlKey is nullable
-        if ($this->network && $this->network->getUrlKey()) {
-            return $this->network->getUrlKey();
-        }
-        return '';
+        $urlKey = $this->network ? $this->network->getUrlKey() : null;
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Section/Footer/FooterPresenter.php:35]: If you're making this into a ternary why not just do
```suggestion
        return $this->network ? $this->network->getUrlKey() : '';
```
?

Comment (BjornTwachtmann) [reply src/Ds2013/Presenters/Section/Footer/FooterPresenter.php:35]: Actually, ignore that, I'm an idiot. I think this read better before though tbh.

File: src/Ds2013/Presenters/Domain/BroadcastEvent/broadcast_event.html.twig (hunk 1 above)
Comment (gmarinov) [reply src/Ds2013/Presenters/Domain/BroadcastEvent/broadcast_event.html.twig:36]: I’ll add it for clarity, though `network_link(net)` already returns falsy

Comment (BjornTwachtmann) [reply src/Ds2013/Presenters/Domain/BroadcastEvent/broadcast_event.html.twig:36]: if it's not needed then don't add it, just assumed the check was needed as it was there before. It does look like network_link accepts null, so maybe not necessary

File: tests/Ds2013/Presenters/Domain/BroadcastEvent/BroadcastEventPresenterTest.php [hunk 3]
```diff
@@ -138,6 +130,26 @@ public function testGetFirstNetworkWhenDoesntExistNetworks()
...
+    {
+        $routeCollectionBuilder = new RouteCollectionBuilder();
+        $routeCollectionBuilder->add('/{networkUrlKey}', '', 'network');
+        $router = new UrlGenerator($routeCollectionBuilder->build(), new RequestContext());
+
+        // Radio network should not link to NHP because radio hasn't
+        [, , $broadcast] = $this->buildCBroadcastsWithSpecificNetwork('bbc_radio_one', 'radio1', 'radio 1', NetworkMediumEnum::RADIO);
+        $presenter = $this->buildPresenterForBroadcast($broadcast, $router);
+        $c = $this->presenterCrawler($presenter);
+        $this->assertEquals($c->filter('a')->count(), 0, 'network should not be linked');
```
Comment (Tasiobg) [inline tests/Ds2013/Presenters/Domain/BroadcastEvent/BroadcastEventPresenterTest.php:143]: Small thing, the order of arguments for "assertEquals" is $exptected, $actual, $message
I think the right order can help reading test errors.

File: tests/Ds2013/Presenters/Domain/BroadcastEvent/BroadcastEventPresenterTest.php [hunk 4]
```diff
@@ -195,9 +207,25 @@ private function getListServicesNotDoingLiveBroadcasts(): array
...
+            ],
+            'medium' => $medium,
+            'name' => $name,
+            'urlKey' => $urlKey,
+            'nid' => new Nid($nidStr),
+        ])->build();
+        return $this->buildCBroadcastsWithNetworksAndServices($network);
+    }
+
+    private function buildCBroadcastsWithNetworksAndServices($specificNetwork = null) :array
```
Comment (Tasiobg) [inline tests/Ds2013/Presenters/Domain/BroadcastEvent/BroadcastEventPresenterTest.php:226]:
```suggestion
    private function buildCBroadcastsWithNetworksAndServices($specificNetwork = null): array
```

File: tests/Ds2013/Presenters/Domain/BroadcastEvent/BroadcastEventPresenterTest.php [hunk 5]
```diff
@@ -14,33 +14,27 @@
 use BBC\ProgrammesPagesService\Domain\Entity\CollapsedBroadcast;
 use BBC\ProgrammesPagesService\Domain\Entity\Network;
 use BBC\ProgrammesPagesService\Domain\Entity\Service;
+use BBC\ProgrammesPagesService\Domain\Enumeration\NetworkMediumEnum;
+use BBC\ProgrammesPagesService\Domain\ValueObject\Nid;
 use BBC\ProgrammesPagesService\Domain\ValueObject\Sid;
-use PHPUnit\Framework\TestCase;
+use Symfony\Component\DomCrawler\Crawler;
```
Comment (Tasiobg) [inline tests/Ds2013/Presenters/Domain/BroadcastEvent/BroadcastEventPresenterTest.php:20]:
```suggestion

```

File: tests/Ds2013/Presenters/Section/Footer/FooterPresenterTest.php [hunk 6]
```diff
@@ -10,31 +10,39 @@
...
-
-        $navLinks = ['title' => 'Schedule', 'url' => 'path/to/schedule'];
-        $mockNetwork->method('getOption')->with('navigation_links')->willReturn($navLinks);
-
-        $mockNetwork->method('getNid')->willReturn(new Nid('bbc_one'));
-        $mockNetwork->method('getImage')->willReturn($mockImage);
-
-        $this->mockProgramme = $this->createMock(Programme::class);
-        $this->mockProgramme->method('getNetwork')->willReturn($mockNetwork);
+        $mockProgramme = function ($nid, $networkName, $netUrlKey, $medium) : PHPUnit_Framework_MockObject_MockObject {
```
Comment (Tasiobg) [inline tests/Ds2013/Presenters/Section/Footer/FooterPresenterTest.php:28]:
```suggestion
        $mockProgramme = function ($nid, $networkName, $netUrlKey, $medium): PHPUnit_Framework_MockObject_MockObject {
```

File: tests/DsAmen/Presenters/Section/Footer/FooterPresenterTest.php [hunk 7]
```diff
@@ -9,32 +9,40 @@
 use BBC\ProgrammesPagesService\Domain\Entity\Image;
 use BBC\ProgrammesPagesService\Domain\Entity\Network;
 use BBC\ProgrammesPagesService\Domain\Entity\Programme;
+use BBC\ProgrammesPagesService\Domain\Enumeration\NetworkMediumEnum;
 use BBC\ProgrammesPagesService\Domain\ValueObject\Nid;
-use PHPUnit\Framework\TestCase;
+use Symfony\Component\DomCrawler\Crawler;
```
Comment (Tasiobg) [inline tests/DsAmen/Presenters/Section/Footer/FooterPresenterTest.php:14]:
```suggestion

```

File: tests/BaseWebTestCase.php [hunk 8]
```diff
@@ -2,6 +2,7 @@
 declare(strict_types = 1);
 namespace Tests\App;
 
+use App\DsShared\BasePresenter;
```
Comment (Tasiobg) [inline tests/BaseWebTestCase.php:5]: Is this line required?

File: tests/Ds2013/Presenters/Domain/BroadcastEvent/BroadcastEventPresenterTest.php (hunk 3 above)
Comment (gmarinov) [reply tests/Ds2013/Presenters/Domain/BroadcastEvent/BroadcastEventPresenterTest.php:143]: my bad..

File: tests/BaseWebTestCase.php (hunk 8 above)
Comment (gmarinov) [reply tests/BaseWebTestCase.php:5]: no. nice catch. thanks.

**Pull request merged**